test/e2e/e2e_test.go

```diff
@@ -35,13 +35,16 @@ import (
 	_ "github.com/intel/pmem-csi/test/e2e/tls"
 
 	"github.com/intel/pmem-csi/test/e2e/deploy"
+	"github.com/intel/pmem-csi/test/e2e/operator"
 )
 
 func TestMain(m *testing.M) {
 	klog.SetOutput(GinkgoWriter)
+	logs.InitLogs()
+
+	operator.DefineTests()
```

Comment:
Can the operator tests co-exist with an installed PMEM instance from `deploy`? If not, then you have to ensure that the other instance gets removed first.

You could do that by adding a "none" deployment name to https://github.com/intel/pmem-csi/blob/d03cf74be569b333e274c2aaee5f732c0b49d45d/test/e2e/deploy/deploy.go#L364-L381 and adding a `DriverRunning` flag that is false for it (and only it).

Then `DescribeForAll` could filter out those deployments which don't have a running driver.

That's the smaller change. The bigger change would be to also create a "operator" deployment which has `DriverRunning = false` and a new `OperatorRunning = true`.

In https://github.com/intel/pmem-csi/blob/d03cf74be569b333e274c2aaee5f732c0b49d45d/test/e2e/deploy/deploy.go#L387 then deploy the operator instead of the driver.

`RemovePMEMDriver` needs to be extended accordingly.

That way, the entire infrastructure for registering operator tests becomes unnecessary.

Long-term we then can add another deployment which first deploys the operator and the driver through the operator and run all of the normal tests against that, if we decide that this is worthwhile.

Reply:
> Can the operator tests co-exist with an installed PMEM instance from deploy?

Yes, they supposed to co-exists and they shouldn't disturb the existing driver instances. The idea here is the operator tests deploys the new driver instances in their own namespaces and uses fake driver images so that they do not corrupt underlined pmem space.

Reply:
Is this explained somewhere?

Reply:
I think it would be simpler to deploy the real driver as explained in https://github.com/intel/pmem-csi/pull/559#discussion_r404910555. That then makes it possible to write real tests for the behavior of the resulting driver installation.

Reply:
> Is this explained somewhere?

Added a comment in the test code:  https://github.com/intel/pmem-csi/pull/559/files#diff-bed6d8494d52c53c8405f39732a27b2bR42-R45

Reply:
> I think it would be simpler to deploy the real driver as explained in #559 (comment). That then makes it possible to write real tests for the behavior of the resulting driver installation.

My idea of running real tests agint the driver deployed by the operator is by running driver tests after adding new deploy mentod to `deploy/deploy.go`.

Reply:
I'm probably not explaining well what I have in mind. Let's merge this and then I can show it by coding it up.

Reply:
> Added a comment in the test code: https://github.com/intel/pmem-csi/pull/559/files#diff-bed6d8494d52c53c8405f39732a27b2bR42-R45

Beware that such deep links for diffs tend to break during rebasing (if it ever worked at all); this one doesn't highlight the actual change anymore. But I found what you added.